File: AoC_01_Entrance.cs
var data = File.ReadAllLines(@"input.txt");

int dial = 50;
int zero = 0;
// --- PART 1
foreach (var line in data)
{
    int rotation = int.Parse(line[1..]);
    dial = (dial +
        (line[0] == 'L' ? (100 - rotation) : rotation)) % 100;

    if (dial == 0)
        zero++;
}
Console.WriteLine(zero);
dial = 50;
zero = 0;

// --- PART 2

foreach (var line in data)
{
    int rotation = int.Parse(line[1..]);

    while (rotation != 0)
    {
        if (line[0] == 'L')
            dial--;
        else
            dial++;
        dial = dial % 100;
        if (dial == 0)
            zero++;
        rotation--;
    }
}

Console.WriteLine(zero);

File: AoC_01_Entrance.zig
const std = @import("std");

pub fn main() !void {
    try part_one();
    try part_two();
}

pub fn part_one() !void {
    const file = try std.fs.cwd().openFile("input.txt", .{});
    defer file.close();

    var buffer: [100]u8 = undefined;
    var dial: i32 = 50;
    var zero: i32 = 0;
    var file_reader = file.reader(&buffer);
    while (file_reader.interface.takeDelimiterExclusive('\n')) |line| {
        const rotation: i32 = try std.fmt.parseInt(i32, line[1..], 10);

        dial = @mod(dial +
            (if (line[0] == 'L') (100 - rotation) else rotation), 100);
        if (dial == 0) zero = zero + 1;

        if (!file_reader.atEnd())
            file_reader.interface.toss(1);
    } else |_| {}
    std.debug.print("{d}\n", .{zero});
}

pub fn part_two() !void {
    const file = try std.fs.cwd().openFile("input.txt", .{});
    defer file.close();

    var buffer: [100]u8 = undefined;
    var dial: i32 = 50;
    var zero: i32 = 0;
    var file_reader = file.reader(&buffer);
    while (file_reader.interface.takeDelimiterExclusive('\n')) |line| {
        var rotation: i32 = try std.fmt.parseInt(i32, line[1..], 10);

        while (rotation != 0) {
            const increment: i32 = if (line[0] == 'L') (-1) else (1);
            dial = @mod(dial +
                increment, 100);
            if (dial == 0) {
                zero = zero + 1;
            }
            rotation = rotation - 1;
        }

        if (!file_reader.atEnd())
            file_reader.interface.toss(1);
    } else |_| {}
    std.debug.print("{d}\n", .{zero});
}
